content/doc/book/pipeline/getting-started.adoc

```diff
@@ -0,0 +1,233 @@
+link:http://groovy-lang.org/semantics.html[Groovy syntax]
+will be introduced as necessary in this document, so while an understanding of
+Groovy is helpful, it is not required to use Pipeline Script.
+
+A basic Pipeline can be created in either of the following ways:
+
+* By entering a script directly in the Jenkins web UI.
+* By creating a `Jenkinsfile` which can be checked into a project's source
+  control repository.
+
+The syntax for defining a Pipeline with either approach is the same, but while
+Jenkins supports entering Pipeline Script directly into the web UI, it's
+generally considered best practice to define the Pipeline in a `Jenkinsfile`
+which Jenkins can then load directly from source control.
+footnoteref:[scm, https://en.wikipedia.org/wiki/Source_control_management]
+
+
+=== Defining a Pipeline in the Web UI
+
+To create a basic Pipeline in the Jenkins web UI, follow these steps:
+
+* Click *New Item* on Jenkins home page.
+
+image::new-item-selection.png["Click *New Item* on the Jenkins home page", role=center]
+
+* Enter a name for your Pipeline, select *Pipeline* and click *OK*.
+
+[CAUTION]
+====
+Jenkins uses the name of the Pipeline to create directories on disk. Pipeline
+names which include spaces may uncover bugs in scripts which do not expect
+paths to contain spaces.
+====
+
+image::new-item-creation.png["Enter a name, select *Pipeline*, and click *OK*", role=center]
+
+
+
+* In the *Script* text-area, enter a Pipeline script and click *Save*.
+
+image::hello-world-script.png["In the *Script* text-area, enter a Pipeline script and click Save", role=center]
+
+* Click *Build Now* to run the Pipeline.
+
+image::build-now.png["Click *Build Now* to run the Pipeline", role=center]
+
+
+* Click *#1* under "Build History" and then click *Console Output* to see the
+  full output from the Pipeline.
+
+image::hello-world-console-output.png["Click *Console Output* for the Pipeline", role=center]
+
+The example above shows a successful run of a basic Pipeline created in the Jenkins
+web UI, using two valuable steps.
+
+[pipeline]
+----
+// Script //
+node { // <1>
+    echo 'Hello World' // <2>
+}
+// Declarative not yet implemented //
+----
+<1> `node` allocates an executor and workspace in the Jenkins environment.
+<2> `echo` writes simple string in the Console Output.
+
+
+// Despite :sectanchors:, explicitly defining an anchor because it will be
+// referenced from other documents
+[[defining-a-pipeline-in-scm]]
+=== Defining a Pipeline in SCM
+
+Complex pipelines would be cumbersome to write and maintain if you could only do
+that in the text area provided by the Jenkins job configuration page.
+
+Accordingly, you also have the option of writing pipeline scripts (Jenkinsfiles)
+with the editor that you use in your IDE (integrated development environment) or
+SCM system, and then loading those scripts into Jenkins using the *Pipeline
+Script from SCM* option enabled by the workflow-scm-step plugin, which is one of
```

Comment:
If workflow-scm-step plugin is automatically loaded, do we need to mention it here?